File: benchmarks/node/json_roundtrip.js
let result = 0;
for (let i = 0; i < 100; i++) {
    const obj = {};
    for (let k = 0; k < 1000; k++) obj[String(k)] = k;
    const s = JSON.stringify(obj);
    const parsed = JSON.parse(s);
    result = parsed["999"];
}
console.log(result);

File: benchmarks/go/json_roundtrip.go
package main

import (
	"encoding/json"
	"fmt"
)

func main() {
	for iter := 0; iter < 100; iter++ {
		// build map with 1k keys
		m := make(map[string]int, 1000)
		for i := 0; i < 1000; i++ {
			key := fmt.Sprintf("key%d", i)
			m[key] = i
		}

		// stringify
		data, err := json.Marshal(m)
		if err != nil {
			panic(err)
		}

		// parse
		var out map[string]int
		if err := json.Unmarshal(data, &out); err != nil {
			panic(err)
		}
	}
	fmt.Println(999)
}
